alt_installer.py

```diff
@@ -0,0 +1,756 @@
+            evt='<<GetLogos>>'
+        )
+        verified = utils.verify_downloaded_file(
+            config.LOGOS64_URL,
+            downloaded_file,
+            app=app,
+            evt='<<CheckLogos>>'
+        )
+        if not verified:
+            msg.logos_error(f"File verification  failed: {downloaded_file}")
+    # Copy file into INSTALLDIR.
+    installer = Path(f"{config.INSTALLDIR}/data/{config.LOGOS_EXECUTABLE}")
+    if not installer.is_file():
+        shutil.copy(downloaded_file, installer.parent)
+
+    logging.debug(f"> '{str(installer)}' exists?: {installer.is_file()}")
+
+
+def ensure_wineprefix_init(app=None):
+    config.INSTALL_STEPS_COUNT += 1
+    ensure_product_installer_filecheck(app=app)
+    config.INSTALL_STEP += 1
+    update_install_feedback("Ensuring wineprefix is initialized…", app=app)
+
+    init_file = Path(f"{config.WINEPREFIX}/system.reg")
+    if not init_file.is_file():
+        wine.initializeWineBottle()
+    logging.debug(f"> {init_file} exists?: {init_file.is_file()}")
+
+
+def ensure_winetricks_applied(app=None):
+    config.INSTALL_STEPS_COUNT += 1
+    ensure_wineprefix_init(app=app)
+    config.INSTALL_STEP += 1
+    status = "Ensuring winetricks & other settings are applied…"
+    update_install_feedback(status, app=app)
+    logging.debug('- disable winemenubuilder')
+    logging.debug('- settings renderer=gdi')
+    logging.debug('- corefonts')
+    logging.debug('- tahoma')
+    logging.debug('- settings fontsmooth=rgb')
+    logging.debug('- d3dcompiler_47')
+
+    usr_reg = Path(f"{config.WINEPREFIX}/user.reg")
+    sys_reg = Path(f"{config.WINEPREFIX}/system.reg")
+    if not grep(r'"winemenubuilder.exe"=""', usr_reg):
+        reg_file = os.path.join(config.WORKDIR, 'disable-winemenubuilder.reg')
+        with open(reg_file, 'w') as f:
+            f.write(r'''REGEDIT4
+
+[HKEY_CURRENT_USER\Software\Wine\DllOverrides]
+"winemenubuilder.exe"=""
+''')
+        wine.wine_reg_install(reg_file)
+
+    if not grep(r'"renderer"="gdi"', usr_reg):
+        wine.winetricks_install("-q", "settings", "renderer=gdi")
+
+    if not config.SKIP_FONTS and not grep(r'"Tahoma \(TrueType\)"="tahoma.ttf"', sys_reg):  # noqa: E501
+        wine.installFonts()
+
+    if not grep(r'"\*d3dcompiler_47"="native"', usr_reg):  # noqa: E501
+        wine.installD3DCompiler()
+
+    if not grep(r'"ProductName"="Microsoft Windows 10"', sys_reg):
+        if not config.WINETRICKS_UNATTENDED:
+            wine.winetricks_install("-q", "settings", "win10")
+        else:
+            wine.winetricks_install("settings", "win10")
+    logging.debug("> Done.")
+
+
+def ensure_product_installed(app=None):
+    config.INSTALL_STEPS_COUNT += 1
+    ensure_winetricks_applied(app=app)
+    config.INSTALL_STEP += 1
+    update_install_feedback("Ensuring product is installed…", app=app)
+
+    if not utils.find_installed_product():
+        wine.install_msi()
+        config.LOGOS_EXE = utils.find_installed_product()
```

Comment:
This step hung for 7 minutes before I cancelled it.

I restarted the install for a third time through—definitely nice to skip over steps like fonts—and the installer again hung at this function for 4 mins before cancelling again.

Reply:
I can't recreate this. I've run multiple installations by first removing the INSTALLDIR and the config file, as well as running the installer with the app already installed. Are we using different parameters? I'm using Logos 10 v29.1.0.0022, recommended appimage, system winetricks, don't install fonts, passive install of app, so `./LogosLinuxInstaller.py -PDF`.

Reply:
I did not use -P, but the same otherwise.